zoom_rooms/indico_zoom_rooms/tasks.py

```diff
@@ -0,0 +1,130 @@
+# This file is part of the CERN Indico plugins.
+# Copyright (C) 2024 CERN
+#
+# The CERN Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License; see
+# the LICENSE file for more details.
+
+from pprint import pformat
+
+import requests
+from celery.schedules import crontab
+from celery.utils.log import get_task_logger
+
+from indico.core.celery import celery
+from indico.core.db import db
+from indico.modules.events.contributions.models.contributions import Contribution
+from indico.modules.events.models.events import Event
+from indico.modules.events.sessions.models.blocks import SessionBlock
+from indico.util.string import strip_control_chars
+
+from indico_zoom_rooms.models import ZoomRoomsAction, ZoomRoomsQueueEntry
+
+
+def obj_from_locator(locator: dict) -> Contribution | SessionBlock | Event:
+    """Convert a locator dict into a usable DB object."""
+    if 'contrib_id' in locator:
+        return Contribution.get(locator['contrib_id'])
+    elif 'block_id' in locator:
+        return SessionBlock.get(locator['block_id'])
+    elif 'event_id' in locator:
+        return Event.get(locator['event_id'])
+    else:
+        raise ValueError("Can't decode locator")
+
+
+logger = get_task_logger(__name__)
```

Comment:
I would rather use the Indico loggers for it, e.g. `current_plugin.logger` if this runs in the plugin context which I think it does. We want those log messages in `indico.log` and not `celery.log`

Reply:
Gotcha. Done.